src/query/plan/preprocess.hpp

```diff
@@ -200,7 +205,7 @@ class PatternFilterVisitor : public ExpressionVisitor<void> {
   void Visit(NamedExpression &op) override{};
   void Visit(RegexMatch &op) override{};
 
-  std::vector<FilterMatching> getMatchings() { return matchings_; }
+  std::vector<FilterMatching> getMatchings();
```

Comment:
Note from Gareth. We could just move the declaration down in the hpp, instead of moving it to the cpp.

Reply:
This also applies to other constructor/desctuctors in this hpp

Reply:
won't work, everything which is defaulted / custom defined -> moved under cpp file